plugins/lsp/language-servers/pylsp.nix

```diff
@@ -536,14 +536,28 @@ in
           mapAttrsToList
             (
               pluginName: nixPackage:
+              let
+                # Get rid of the python-lsp-server dependency
+                filterDependencies = dependencies: filter (dep: dep.pname != "python-lsp-server") dependencies;
+              in
               (optional (isEnabled plugins.${pluginName}) (
-                nixPackage.overridePythonAttrs (old: {
-                  # Get rid of the python-lsp-server dependency
-                  propagatedBuildInputs = filter (dep: dep.pname != "python-lsp-server") old.propagatedBuildInputs;
-
-                  # Skip testing because those naked dependencies will complain about missing pylsp
-                  doCheck = false;
-                })
+                nixPackage.overridePythonAttrs (
+                  old:
+                  (lib.genAttrs
+                    # NOTE: Depending on the exact nixpkgs rev, we may have `dependencies` or `propagatedBuildInputs`
+                    # See https://github.com/NixOS/nixpkgs/pull/327630#discussion_r1679253679
+                    # TODO: Only filter `dependencies`
+                    (lib.filter (n: old ? ${n}) [
+                      "propagatedBuildInputs"
+                      "dependencies"
+                    ])
+                    (n: filterDependencies old.${n})
```

Comment:
Since this is done gracefully I don't see why we can't merge now?

Reply:
If you are talking about merging the PR, tests were failing. I just merged `main` to here and tests are green again

Reply:
Our CI is currently disabled, so the only "test" running on GitHub is `mergify` (which doesn't actually test any code).

Or are you referring to tests run locally?

Reply:
I referred to [this test failure](https://buildbot.nix-community.org/#/builders/43/builds/287) in 6066ff2874c3b12a2418d91ca508e18340585276, and local tests fail for me even on main with some huge and totally unrelated error.

Reply:
> I referred to [this test failure](https://buildbot.nix-community.org/#/builders/43/builds/287) in [6066ff2](https://github.com/nix-community/nixvim/commit/6066ff2874c3b12a2418d91ca508e18340585276),

The linked failure is #1878 (one of the reasons CI is currently disabled).

> local tests fail for me even on main with some huge and totally unrelated error.

Strange. I've just run the test suite locally on another PR and everything passed just fine...

Are you using the `tests` devshell command or just `nix flake check`? Are you trying to use `--all-systems`?

Reply:
Just ran `tests` on this branch, and everything passes for me.

Reply:
I used `nix flake check --all-systems -v` and got these logs:

[logs.txt.gz](https://github.com/user-attachments/files/16338335/logs.txt.gz)
(too big for pastebin, so sending as a file)

Anyway, this doesn't really matter. If all tests are passing, why not merge this PR?

Reply:
> why not merge this PR?

As mentioned above, I'm delegating to @GaetanLepage here since he's dealing with these changes in nixpkgs.